foyer-memory/benches/bench_hit_ratio.rs

```diff
@@ -0,0 +1,201 @@
+
+use ahash::RandomState;
+use foyer_memory::{
+    cache::{LfuCache, LruCache},
+    eviction::Eviction,
+    indexer::Indexer,
+    CacheEventListener, DefaultCacheEventListener, FifoCache, FifoCacheConfig, FifoConfig, GenericCache, Handle,
+    LfuCacheConfig, LfuConfig, LruCacheConfig, LruConfig,
+};
+use rand::{distributions::Distribution, thread_rng};
+
+type CacheKey = String;
+type CacheValue = ();
+
+const ITEMS: usize = 10_000;
+const ITERATIONS: usize = 5_000_000;
+
+const SHARDS: usize = 1;
+const OBJECT_POOL_CAPACITY: usize = 16;
+/*
+inspired by pingora/tinyufo/benches/bench_hit_ratio.rs
+cargo bench --bench bench_hit_ratio
+
+zif_exp, cache_size             fifo            lru             lfu             moka
+0.90, 0.005                     16.26%          19.22%          32.38%          33.46%
+0.90, 0.01                      22.55%          26.21%          38.55%          37.93%
+0.90, 0.05                      41.10%          45.60%          55.45%          55.26%
+0.90,  0.1                      51.11%          55.72%          63.83%          64.21%
+0.90, 0.25                      66.81%          71.17%          76.21%          77.14%
+1.00, 0.005                     26.64%          31.08%          44.14%          45.61%
+1.00, 0.01                      34.37%          39.13%          50.60%          50.67%
+1.00, 0.05                      54.06%          58.79%          66.79%          66.99%
+1.00,  0.1                      63.12%          67.58%          73.92%          74.38%
+1.00, 0.25                      76.14%          79.92%          83.61%          84.33%
+1.05, 0.005                     32.63%          37.68%          50.24%          51.77%
+1.05, 0.01                      40.95%          46.09%          56.75%          57.15%
+1.05, 0.05                      60.47%          65.06%          72.07%          72.36%
+1.05,  0.1                      68.96%          73.15%          78.52%          78.96%
+1.05, 0.25                      80.42%          83.76%          86.79%          87.42%
+1.10, 0.005                     39.02%          44.52%          56.29%          57.90%
+1.10, 0.01                      47.66%          52.99%          62.64%          63.23%
+1.10, 0.05                      66.60%          70.95%          76.94%          77.25%
+1.10,  0.1                      74.26%          78.09%          82.56%          82.93%
+1.10, 0.25                      84.15%          87.06%          89.54%          90.05%
+1.50, 0.005                     81.19%          85.28%          88.91%          89.94%
+1.50, 0.01                      86.91%          89.87%          92.24%          92.78%
+1.50, 0.05                      94.75%          96.04%          96.95%          97.07%
+1.50,  0.1                      96.65%          97.51%          98.06%          98.15%
+1.50, 0.25                      98.35%          98.81%          99.04%          99.09%
+*/
+fn cache_hit<H, E, I, L, S>(
```

Comment:
To simplify and restore the mod visibility, please use `enum Cache` instead of internal structures or traits.